packages/beacon-node/test/unit/chain/prepareNextSlot.test.ts

```diff
@@ -1,170 +1,163 @@
-    regenStub = sandbox.createStubInstance(QueuedStateRegenerator) as SinonStubbedInstance<QueuedStateRegenerator> &
-      QueuedStateRegenerator;
-    chainStub.regen = regenStub;
-    loggerStub = createStubbedLogger(sandbox);
-    beaconProposerCacheStub = sandbox.createStubInstance(
-      BeaconProposerCache
-    ) as SinonStubbedInstance<BeaconProposerCache> & BeaconProposerCache;
-    (chainStub as unknown as {beaconProposerCache: BeaconProposerCache})["beaconProposerCache"] =
-      beaconProposerCacheStub;
-    getForkStub = sandbox.stub(config, "getForkName");
-    executionEngineStub = sandbox.createStubInstance(ExecutionEngineHttp) as SinonStubbedInstance<ExecutionEngineHttp> &
-      ExecutionEngineHttp;
-    (chainStub as unknown as {executionEngine: IExecutionEngine}).executionEngine = executionEngineStub;
-    (chainStub as unknown as {config: ChainForkConfig}).config = config as unknown as ChainForkConfig;
-    chainStub.opts = {emitPayloadAttributes} as IChainOptions;
+    forkChoiceStub = chainStub.forkChoice;
+    regenStub = chainStub.regen;
+    loggerStub = getMockedLogger();
+    beaconProposerCacheStub = chainStub.beaconProposerCache;
+
+    getForkStub = vi.spyOn(config, "getForkName");
+    executionEngineStub = chainStub.executionEngine;
+    vi.spyOn(chainStub, "opts", "get").mockReturnValue({emitPayloadAttributes} as IChainOptions);
 
     scheduler = new PrepareNextSlotScheduler(chainStub, config, null, loggerStub, abortController.signal);
+
+    vi.spyOn(regenStub, "getBlockSlotState");
   });
 
   afterEach(() => {
-    sandbox.restore();
+    vi.clearAllMocks();
+    vi.clearAllTimers();
   });
 
   it("pre bellatrix - should not run due to not last slot of epoch", async () => {
-    getForkStub.returns(ForkName.phase0);
+    getForkStub.mockReturnValue(ForkName.phase0);
     await scheduler.prepareForNextSlot(3);
-    expect(chainStub.recomputeForkChoiceHead).not.to.be.called;
+    expect(chainStub.recomputeForkChoiceHead).not.toHaveBeenCalled();
   });
 
   it("pre bellatrix - should skip, headSlot is more than 1 epoch to prepare slot", async () => {
-    getForkStub.returns(ForkName.phase0);
-    chainStub.recomputeForkChoiceHead.returns({slot: SLOTS_PER_EPOCH - 2} as ProtoBlock);
+    getForkStub.mockReturnValue(ForkName.phase0);
+    chainStub.recomputeForkChoiceHead.mockReturnValue({slot: SLOTS_PER_EPOCH - 2} as ProtoBlock);
     await Promise.all([
       scheduler.prepareForNextSlot(2 * SLOTS_PER_EPOCH - 1),
-      sandbox.clock.tickAsync((config.SECONDS_PER_SLOT * 1000 * 2) / 3),
+      vi.advanceTimersByTimeAsync((config.SECONDS_PER_SLOT * 1000 * 2) / 3),
     ]);
-    expect(chainStub.recomputeForkChoiceHead, "expect updateHead to be called").to.be.called;
-    expect(regenStub.getBlockSlotState, "expect regen.getBlockSlotState not to be called").not.to.be.called;
+    // "expect updateHead to be called"
+    expect(chainStub.recomputeForkChoiceHead).toHaveBeenCalled();
```

Comment:
I am not sure why there was `"expect updateHead to be called"` as assertion message, maybe the default message of `.to.be.called` was not great. But I'd say adding this as a comment here doesn't is just noise, the line below literally says the same thing.

Note: There are a bunch of places where this is the case, I'd suggest this is cleaned up as otherwise it would advocate for this kind of style of commenting which is just not a good practice.